Support for extra column from pytorch nonzero

nykwil opened the pull request:
Also put a gitignore.

nykwil committed:
while loops should always move towards the exit condition. Fixes a bug when shape size goes into negative.

Also put a gitignore.

nykwil committed:
compensating for torch.nonzero returning an empty column

nykwil commented:
Have a fix for the reason this occurred after torch update.

JoeNavark commented:
It does run but the behavior is not completely as expected. Cropped image is not centered on mask, and sometimes does not even contain the masked area.

nykwil commented:
> It does run but the behavior is not completely as expected. Cropped image is not centered on mask, and sometimes does not even contain the masked area.

The bug is the extra column in nonzero, the change to the while loop is just better style so you don't get infinite loops when you get results you aren't expecting. Those new cases should never run when nonzero is working properly.

nykwil committed:
Update mask_boundingbox.py

support for padding also fix bounding box output

nykwil commented:
Fixed a bug with the bounded mask return which also needs to have the new column if you are using the newer comfy version.

Added a padding feature.

nykwil committed:
Update mask_boundingbox.py

fix another bug with masks that are even larger, fixed the raw bb returns, min sizes uses pads (centered)

nykwil commented:
Fixed another bug if there's another mask format improved min sizes.
